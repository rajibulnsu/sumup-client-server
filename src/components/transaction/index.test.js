import React from 'react';
import { shallow } from 'enzyme';
import Transaction from './index';
import renderer from 'react-test-renderer';

describe('Transaction', () => {
  it('shoult match the snapshot of transaction', () => {
    const item = {
      "transactionId": 684642834690,
      "transactionDate": "2020-01-10T10:02:49.817Z",
      "status": "Successful",
      "amount": 162.45,
      "paymentMethod": "Card"
    };
    const wrapper = renderer.create(<Transaction detail={item}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should include `status__failed` class if status is `Failed`', () => {
    const item = {
      "transactionId": 684642834690,
      "transactionDate": "2020-01-10T10:02:49.817Z",
      "status": "Failed",
      "amount": 162.45,
      "paymentMethod": "Card"
    };
    const wrapper = shallow(<Transaction detail={item} />);

    expect(wrapper.find('.transaction__amount').hasClass('status__failed')).toBe(true);
  });
});