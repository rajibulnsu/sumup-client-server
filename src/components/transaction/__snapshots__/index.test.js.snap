// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction shoult match the snapshot of transaction 1`] = `
<div
  className="transaction__row"
>
  <div
    className="transaction__data_status"
  >
    <span
      className="transaction__date"
    >
      10/01/2020
    </span>
    <span
      className="status successful"
    >
      Successful
    </span>
  </div>
  <div
    className="transaction__amount_type"
  >
    <span
      className="transaction__amount null"
    >
      162.45
       â‚¬
    </span>
    <span
      className="transaction__type"
    >
      <i
        className="far fa-credit-card"
      />
    </span>
  </div>
</div>
`;
